crates/wasmtime/src/store.rs

```diff
@@ -1496,8 +1496,109 @@ unsafe impl<T> wasmtime_runtime::Store for StoreInner<T> {
         (&mut inner.externref_activations_table, &inner.modules)
     }
 
-    fn limiter(&mut self) -> Option<&mut dyn wasmtime_runtime::ResourceLimiter> {
-        <Self>::limiter(self)
+    #[cfg(not(feature = "async"))]
+    fn limiter_memory_growing(
+        &mut self,
+        current: usize,
+        desired: usize,
+        maximum: Option<usize>,
+    ) -> bool {
+        if let Some(limiter) = <Self>::limiter(self) {
+            limiter.memory_growing(current, desired, maximum)
+        } else {
+            true
+        }
+    }
+
+    #[cfg(feature = "async")]
```

Comment:
Ah yeah to expand on some thoughts from above, I don't think we'll want this conditional on the `async` cargo feature since that's on-by-default but rather some runtime configuration in the limiter itself which indicates which version to use.